### PR TITLE
[CARBONDATA-2498] Change CarbonWriterBuilder interface to take schema while creating writer

### PR DESCRIPTION
**Pull request by kunal642**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5947/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4791/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4792/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5948/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5949/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4793/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5950/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4979/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4980/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4981/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4982/

File: examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java [hunk 1]
```diff
@@ -44,11 +44,10 @@ public static void main(String[] args) {
             fields[1] = new Field("age", DataTypes.INT);
 
             CarbonWriter writer = CarbonWriter.builder()
-                    .withSchema(new Schema(fields))
```
Comment (ravipesala) [inline examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java]: Is withSchema is optional earlier or mandatory to provide ?

Comment (kunal642) [reply examples/spark2/src/main/java/org/apache/carbondata/examples/sdk/CarbonReaderExample.java]: mandatory to provide

File: docs/sdk-writer-guide.md [hunk 2]
```diff
@@ -33,9 +33,9 @@ These SDK writer output contains just a carbondata and carbonindex files. No met
  
      Schema schema = new Schema(fields);
  
-     CarbonWriterBuilder builder = CarbonWriter.builder().withSchema(schema).outputPath(path);
```
Comment (ajantha-bhat) [inline docs/sdk-writer-guide.md]: In the same document handle the below points. 
1. remove withSchema() interface 
2. update the below interfaces, 
buildWriterForCSVInput
buildWriterForAvroInput

File: docs/sdk-writer-guide.md [hunk 3]
```diff
@@ -93,9 +93,8 @@ public class TestSdkAvro {
 
     try {
       CarbonWriter writer = CarbonWriter.builder()
-          .withSchema(carbonSchema)
           .outputPath(path)
-          .buildWriterForAvroInput();
+          .buildWriterForAvroInput(new org.apache.avro.Schema.Parser().parse(avroSchema));
```
Comment (ajantha-bhat) [inline docs/sdk-writer-guide.md]: remove these above lines. No need to prepare avro schema now

    // prepare carbon schema from avro schema 
     org.apache.carbondata.sdk.file.Schema carbonSchema =	     org.apache.carbondata.sdk.file.Schema carbonSchema =
             AvroCarbonWriter.getCarbonSchemaFromAvroSchema(avroSchema);	             AvroCarbonWriter.getCarbonSchemaFromAvroSchema(avroSchema);

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4833/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5011/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4836/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5996/

File: docs/sdk-writer-guide.md (hunk 2 above)
Comment (kunal642) [reply docs/sdk-writer-guide.md]: done

File: docs/sdk-writer-guide.md (hunk 3 above)
Comment (kunal642) [reply docs/sdk-writer-guide.md]: removed

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5015/

**Comment by ravipesala** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4842/

**Pull request closed without merging**